vm/systemSmartContracts/delegation.go

```diff
@@ -602,19 +630,24 @@ func (d *delegation) stakeNodes(args *vmcommon.ContractCallInput) vmcommon.Retur
 		return vmcommon.UserError
 	}
 
+	//TODO: Missing this line, before calling saveGlobalFundData?: globalFund.TotalActive.Sub(globalFund.TotalActive, stakeValue)
+	//Or otherwise, calling saveGlobalFundData is not needed as it was nothing modified in the globalFund structure
+
 	err = d.saveGlobalFundData(globalFund)
 	if err != nil {
 		d.eei.AddReturnMessage(err.Error())
 		return vmcommon.UserError
 	}
 
 	stakeArgs := makeStakeArgs(listToCheck, args.Arguments)
+	//TODO: If executeOnAuctionSC will fail, then the last saves done will remain saved. Is this ok?
 	vmOutput, err := d.executeOnAuctionSC(args.RecipientAddr, "stake", stakeArgs, big.NewInt(0))
 	if err != nil {
 		d.eei.AddReturnMessage(err.Error())
 		return vmcommon.UserError
 	}
 	if vmOutput.ReturnCode != vmcommon.Ok {
+		d.eei.AddReturnMessage(vmOutput.ReturnMessage)
```

Comment:
I think this will duplicate the return message actually. as it was already added.

Reply:
This pattern is used also in init method above. I will remove then also from there